app/screens/modal/UpdateApplicationWarningModal.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import styled from 'styled-components';
+import Modal from '../../components/common/Modal';
+import { Button } from '../../basicComponents';
+import { smColors } from '../../vars';
+
+const ButtonsWrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  margin: auto 0 15px 0;
+  padding-top: 30px;
+`;
+
+const Message = styled.pre`
+  font-size: 14px;
+  line-height: 1.33em;
+  word-wrap: break-word;
+  white-space: pre-wrap;
+  overflow-y: auto;
+  margin-top: 15px;
+
+  ul {
+    list-style: none;
+    margin-left: 10px;
+  }
+  li {
+    margin: 10px 0;
+  }
+  li:before {
+    content: '• ';
+    padding: 5px;
+  }
+`;
+
+const UpdateApplicationWarningModal = ({ isOpen, onApprove, onCancel }) => {
+  if (!isOpen) return null;
+
+  return (
+    <Modal header="Update SMAPP" height={380}>
+      <Message>
+        <p>
+          Restarting now is <b>CRITICAL</b> and may impact your node’s
+          performance and rewards.
+        </p>
+        <ul>
+          <li>
+            Click <b style={{ color: smColors.green }}>RESTART NOW</b> to apply
+            the update immediately. Delaying the update could result in
+            potential loss of rewards.
+          </li>
+          <li>
+            Click <b style={{ color: smColors.purple }}>POSTPONE</b> to delay
+            the update. Be aware that postponing may slow down your node’s
+            performance and future rewards.
+          </li>
+        </ul>
+      </Message>
```

Comment:
This seems to be incorrect. @brusherru please correct me if I'm wrong.

- The loss of the rewards might be caused by restarting (if the node is supposed to publish a proposal at this particular time) and not caused by the fact that the user updates immediately or a couple of hours later. 

- Postponing the restart(and upgrade) should not have such a big impact on the node's performance, not to the extent that the user should risk losing rewards to update immediately. If there are any breaking changes in the code affecting significantly the node's performance and network compatibility (and thus missed rewards) then the forced update is supposed to be used (so no choice for the user anyway, no need to display any modal). 

From the grammatical perspective, it's confusing and quite misleading in several aspects.

Additionally, smapp "knows" the current layer and the number of layers for which the node has rewards eligibility. So as Lane suggested, it would be great to display it only when applicable. There might be users not even smeshing or still generating the POS data. If they see the missed rewards warning they will get lost and scared. 
But, we need to also remember about not restarting while publishing ATX and preparing the proof...

If it's too complex and time-consuming for this version, then my suggestion would be:
```suggestion
      <Message>
        <p>
          To finalize the update, Smapp must be restarted. However, verify your eligibility for upcoming layer rewards.  Powering off the node risks missing proposal submissions and losing the rewards.
        </p>
        <ul>
          <li>
            Click <b style={{ color: smColors.green }}>RESTART NOW</b> to apply
            the update immediately.
          </li>
          <li>
            Click <b style={{ color: smColors.purple }}>POSTPONE</b> to delay
            the update. The persistent "restart" button will remind you about the pending update. 
          </li>
        </ul>
      </Message>
```
Please note that I have not yet tested the functionality, please assess whether or not we want to check proactively the node's and network's state to inform the users more  about the consequences (lost a few layers, the whole epoch, etc.)